File: javascript/leetcode/387. First Unique Character in a String.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    const map = new Map(); 
    
    for(let chr of s) {
        
        if(map.has(chr)) {
            map.set(chr, map.get(chr) + 1)
            continue
        }
        
        map.set(chr, 1)
    } 
    
    for(let i = 0; i < s.length; i++) {
        if(map.get(s[i]) === 1) return i;
    }
    
    return -1;
};

File: c++/leetcode/387. First Unique Character in a String.cpp
class Solution
{
public:
  int firstUniqChar(string s)
  {
    map<char, int> mp;

    for (auto ch : s)
    {
      if (mp.count(ch))
      {
        mp[ch] = mp.at(ch) + 1;
        continue;
      }

      mp[ch] = 1;
    }

    for (int i = 0; i < s.size(); i++)
    {
      if (mp[s.at(i)] == 1)
        return i;
    }

    return -1;
  }
};